packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/analytics_client/event_client/queued_item_store/queued_item_store.dart

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'dart:async';
+
+/// Database for storing strings
+abstract class QueuedItemStore {
+  /// Insert an item to the end of the queue.
+  FutureOr<void> add(String string);
+
+  /// Get the first [count] items from the queue.
+  FutureOr<Iterable<QueuedItem>> get(int count);
+
+  /// Delete [items] from the store.
+  FutureOr<void> deleteItems(Iterable<QueuedItem> items);
```

Comment:
Why does deleteItems take in the items and not the keys?

Reply:
To improve abstraction - consuming code shouldn't need to worry about item ids and parse those ids to perform deletes.